# ANALYSIS ON GYEKS QuickFarm
Gyeks QuickFarm is an e-commerce business that sources and delivers locally produced fresh farm produce directly to customers. The objective of this analysis is to derive actionable insights from six provided CSV datasets, which seeks to:
* Identify the most effective marketing channels.
* Determine the most preferred customer payment method.
* Highlight the most in-demand farm products.
* Pinpoint the area with the highest customer segment.


### NOTEBOOK SETUP

In [349]:
# Import Libraries
import pandas as pd 
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns


### LOAD CSV FILES

In [350]:
feedback = pd.read_csv(r"C:\Users\hp\Music\DATA ANALYSIS\blinkit_customer_feedback.csv")

In [351]:
customers = pd.read_csv(r"C:\Users\hp\Music\DATA ANALYSIS\blinkit_customers.csv")

In [352]:
marketing = pd.read_csv(r"C:\Users\hp\Music\DATA ANALYSIS\blinkit_marketing_performance.csv")

In [353]:
order_item =pd.read_csv(r"C:\Users\hp\Music\DATA ANALYSIS\blinkit_order_items.csv")

In [354]:
order = pd.read_csv(r"C:\Users\hp\Music\DATA ANALYSIS\blinkit_orders.csv")

In [355]:
product = pd.read_csv(r"C:\Users\hp\Music\DATA ANALYSIS\blinkit_products.csv")

### DATA EXPLORATION AND UNDERSTANDING

In [356]:
feedback.head()

feedback_id    order_id  customer_id  rating  \
0      2234710  1961864118     30065862       4   
1      5450964  1549769649      9573071       3   
2       482108  9185164487     45477575       3   
3      4823104  9644738826     88067569       4   
4      3537464  5427684290     83298567       3   

                          feedback_text feedback_category sentiment  \
0         It was okay, nothing special.          Delivery   Neutral   
1              The order was incorrect.    App Experience  Negative   
2         It was okay, nothing special.    App Experience   Neutral   
3      The product met my expectations.    App Experience   Neutral   
4  Product was damaged during delivery.          Delivery  Negative   

  feedback_date  
0    2024-07-17  
1    2024-05-28  
2    2024-09-23  
3    2023-11-24  
4    2023-11-20

In [357]:
feedback.shape

(5000, 8)

In [358]:
feedback.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5000 entries, 0 to 4999
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   feedback_id        5000 non-null   int64 
 1   order_id           5000 non-null   int64 
 2   customer_id        5000 non-null   int64 
 3   rating             5000 non-null   int64 
 4   feedback_text      5000 non-null   object
 5   feedback_category  5000 non-null   object
 6   sentiment          5000 non-null   object
 7   feedback_date      5000 non-null   object
dtypes: int64(4), object(4)
memory usage: 312.6+ KB


### DATA CLEANING

#### 1. Checking for Duplicates

In [359]:
feedback.duplicated().sum()

np.int64(0)

In [360]:
customers.duplicated().sum()

np.int64(0)

In [361]:
marketing.duplicated().sum()

np.int64(0)

In [362]:
order.duplicated().sum()

np.int64(0)

In [363]:
product.duplicated().sum()

np.int64(0)

In [364]:
order_item.duplicated().sum()

np.int64(0)

✅Observation: No duplicates records were found in any of the datasets

#### 2. Checking For Missing Values

In [365]:
feedback.isnull().sum()

feedback_id          0
order_id             0
customer_id          0
rating               0
feedback_text        0
feedback_category    0
sentiment            0
feedback_date        0
dtype: int64

In [366]:
customers.isnull().sum()

customer_id          0
customer_name        0
email                0
phone                0
address              0
area                 0
pincode              0
registration_date    0
customer_segment     0
total_orders         0
avg_order_value      0
dtype: int64

In [367]:
marketing.isnull().sum()

campaign_id          0
campaign_name        0
date                 0
target_audience      0
channel              0
impressions          0
clicks               0
conversions          0
spend                0
revenue_generated    0
roas                 0
dtype: int64

In [368]:
order_item.isnull().sum()

order_id      0
product_id    0
quantity      0
unit_price    0
dtype: int64

In [369]:
order.isnull().sum()

order_id                  0
customer_id               0
order_date                0
promised_delivery_time    0
actual_delivery_time      0
delivery_status           0
order_total               0
payment_method            0
delivery_partner_id       0
store_id                  0
dtype: int64

In [370]:
product.isnull().sum()

product_id           0
product_name         0
category             0
brand                0
price                0
mrp                  0
margin_percentage    0
shelf_life_days      0
min_stock_level      0
max_stock_level      0
dtype: int64

✅Observation: No missing values were detected accross all datasets

#### 3. Updating product_name to match with Gyeks QuickFarm branding

In [371]:
product_copy = product.copy()

In [372]:
# delete all values in product_name
product_copy['product_name']=None

In [373]:
# rename values
crops =[
"Maize","Rice","Millet", "oziza leave","Wheat","Barley","Cassava", "Yam", "Sweet Potato", 
    "Irish Potato", "Cocoyam", "Ginger", "Turmeric", "Groundnut", "Soybean","Cowpea (Beans)", 
    "Bambara Nut", "Sesame Seed", "Sunflower",  "Castor Seed", "Plantain", "Banana", "Pineapple", 
    "Mango", "Orange", "Pawpaw (Papaya)","Guava", "Cashew", "Coconut", "Avocado","Watermelon",
    "Tangerine", "Grapefruit", "Lemon", "Lime", "Tomato", "Pepper", "Onion",  "Okra", "Spinach", 
    "Amaranth leaves", "Fluted Pumpkin (Ugu)", "Bitterleaf",  "Scent  Leaf (Basil)", "Waterleaf",
    "Garden Egg (Eggplant)", "Cabbage", "Lettuce", "Cucumber", "Carrot", "Celosia (Shoko)"
]

In [374]:
print(len(crops))

51


In [375]:
print(crops)

['Maize', 'Rice', 'Millet', 'oziza leave', 'Wheat', 'Barley', 'Cassava', 'Yam', 'Sweet Potato', 'Irish Potato', 'Cocoyam', 'Ginger', 'Turmeric', 'Groundnut', 'Soybean', 'Cowpea (Beans)', 'Bambara Nut', 'Sesame Seed', 'Sunflower', 'Castor Seed', 'Plantain', 'Banana', 'Pineapple', 'Mango', 'Orange', 'Pawpaw (Papaya)', 'Guava', 'Cashew', 'Coconut', 'Avocado', 'Watermelon', 'Tangerine', 'Grapefruit', 'Lemon', 'Lime', 'Tomato', 'Pepper', 'Onion', 'Okra', 'Spinach', 'Amaranth leaves', 'Fluted Pumpkin (Ugu)', 'Bitterleaf', 'Scent  Leaf (Basil)', 'Waterleaf', 'Garden Egg (Eggplant)', 'Cabbage', 'Lettuce', 'Cucumber', 'Carrot', 'Celosia (Shoko)']


In [376]:
# assign new values to product_name
product_copy['product_name'] = np.random.choice(crops, size=len(product_copy))

In [377]:
product = product_copy

### EXPLORATORY DATA ANALYSIS

#### Objective 1: To identify and focus on the most effective marketing channels for better customer targeting to improve sales and conversion rate
* Which channels have the hightest number of clicks?
* Which channels have the highest conversion?
* Which channels generated the highest revenue?


#### Channels with the highest clicks

In [378]:
# sum of clicks per channels

marketing.groupby('channel')[['clicks']].sum().sort_values(by= 'clicks', ascending=False).reset_index()

channel  clicks
0         Email  756057
1           SMS  744274
2  Social Media  738839
3           App  734975

✅Observation/Insight: Email generated the highest number of clicks, indicating strong initial enegagement with customers.

#### Channels with the highest conversions

In [379]:
# sum of conversions per channels
marketing.groupby('channel')[['conversions']].sum().sort_values(
    by= 'conversions',ascending=False).reset_index() 

channel  conversions
0           App        75192
1  Social Media        74940
2         Email        74671
3           SMS        73235

✅Observation/Insight: App channel achieved the highest conversion rate among all channels, indicating that customers engaging through the mobile application are most likely to complete a purchase.

#### Channel with the most generated revenue

In [380]:
# sum of revenue generated per channels
marketing.groupby('channel')[['revenue_generated']].sum().sort_values(
    by = 'revenue_generated', ascending=False).reset_index()

channel  revenue_generated
0         Email         8189331.58
1           App         8075010.49
2  Social Media         7990415.98
3           SMS         7938649.32

✅Observation/Insight: Email channel generated the highest revenue among all marketing channels, contributing the largest share of overall sales.

#### OBJECTIVE 2: To streamline and offer preferred payment methods to improve user experience
* What is the most preferred payment methods by customers?
* What is the most frequently used payment methods?

#### Most preferred payment methods by customers

In [381]:
payment = customers.merge(order, on ='customer_id',how ='right')
# merged customers table with order table

In [382]:
# number of unique customer per payment method
payment.groupby('payment_method')[['customer_id']].nunique().reset_index()

payment_method  customer_id
0           Card         1022
1           Cash          975
2            UPI          972
3         Wallet         1001

#### Most frequently used payment method

In [402]:
# count of payment method per order
payment.groupby('payment_method')[['order_id']].count().reset_index()


payment_method  order_id
0           Card      1285
1           Cash      1257
2            UPI      1214
3         Wallet      1244

✅Observation/Insight: This analysis indicates that Card is the most frequently used method suggesting that Card is the most convenient option, and enhancing support for this method could improve customer satifaction

#### Objective 3: To identify and target areas with high activities to direct sales teams and distribution to where they have the greatest impact and use marketing strategies to revitalize low performing areas to generate high ROI and achieve a stronger market presence
* What areas have the highest number of premiun customers?
* What areas have the highest number of regular customers?
* What areas have the highest number of inactive customers?
* What areas have the highest number of orders?


#### Areas with the highest premium customers

In [384]:
# merged order table with customers table
business_area = order.merge(customers, on ='customer_id',how ='inner')

In [399]:
# keep only premium customers in customer_segment column
premium =business_area[business_area['customer_segment'] == 'Premium']

In [386]:
premium.groupby('area')[['customer_id']].nunique().sort_values(by= 'customer_id', ascending=False).reset_index()

area  customer_id
0                Etawah            6
1                Purnia            6
2             Burhanpur            6
3                  Kota            6
4                Ratlam            5
..                  ...          ...
260  Thiruvananthapuram            1
261            Thrissur            1
262          Vijayawada            1
263          Ahmednagar            1
264               Adoni            1

[265 rows x 2 columns]

✅Observation/Insight: This analysis shows that Etawah, Purnia, Burhanpur and Kota have the highest number of premium customers with 6 customers per area

#### Areas with the highest Regular customers

In [387]:
regular = business_area[business_area['customer_segment']=='Regular']

In [398]:
# keep only regular customers in customer_segment column
regular.groupby('area')[['customer_id']].nunique().sort_values(by='customer_id', ascending=False).reset_index()

area  customer_id
0           Bhagalpur            6
1          Aurangabad            6
2          Pallavaram            6
3           Karaikudi            6
4             Deoghar            6
..                ...          ...
252  Sultan Pur Majra            1
253    Tadepalligudem            1
254         Tadipatri            1
255            Tenali            1
256             Thane            1

[257 rows x 2 columns]

✅Observation/Insight: Bhagalpur,Aurangabad, Pallavaram, Karaikudi and Deoghar all recorded the highest count of Regular customers with 6 customers per area.

#### Areas with the highest inactive customers

In [400]:
# keep only inactive customers in customer_segment column
inactive = business_area[business_area['customer_segment']=='Inactive']

In [390]:
inactive.groupby('area')[['customer_id']].nunique().sort_values(by= 'customer_id', ascending=False).reset_index()

area  customer_id
0          Ratlam            9
1          Kadapa            6
2           Korba            6
3         Deoghar            6
4     Gandhinagar            5
..            ...          ...
236  Tiruvottiyur            1
237      Tinsukia            1
238      Vadodara            1
239    Vijayawada            1
240   Yamunanagar            1

[241 rows x 2 columns]

✅Observation/Insight: Ratlam recorded 9 inactive customers as the highest followed by Kadapa, Korba and Deoghar with a count of 6 customers per area.
This insight can inform targeted marketing campaigns and resource allocation to maximize engagement in high value regions.

#### Areas with the highest orders

In [401]:
# count of order per area
business_area.groupby('area')[['order_id']].count().sort_values(by = 'order_id', ascending = False).reset_index()

area  order_id
0           Orai        44
1        Deoghar        40
2    Gandhinagar        37
3        Nandyal        36
4         Ratlam        35
..           ...       ...
311    Jalandhar         4
312      Cuttack         4
313      Raichur         3
314       Bokaro         2
315    Mangalore         1

[316 rows x 2 columns]

✅Oservation/Insight: The analysis shows Orai recorded the highest number of customer orders, accounting for 40 orders in total.
This suggest a strong customer base and demand concentration in Orai. Targeted marketing and efficient logistics support in this area could further boost sales for Gyeks QuickFarm

#### Objective 4: To identify the most demanded product to meet product demand effectively and prioritize stock availability
* What is the most demanded product?
* What are the top products with the highest number of unique customers?

#### Most demanded products

In [403]:
# merged product table with order-item
product_info = product.merge(order_item, on= 'product_id',how ='inner')

In [404]:
#merged product_info with order table
product_sales = product_info.merge(order, on= 'order_id', how ='inner')

In [405]:
#count of order per product
product_sales.groupby('product_name')[['order_id']].count().sort_values(by= 'order_id', ascending =False).reset_index()

product_name  order_id
0                   Maize       236
1               Tangerine       201
2                Cucumber       182
3             Sesame Seed       172
4                  Tomato       157
5                    Lime       150
6               Pineapple       150
7                 Avocado       150
8                  Banana       145
9         Celosia (Shoko)       138
10              Sunflower       138
11                  Guava       135
12                Coconut       131
13              Waterleaf       121
14                Lettuce       115
15                Cocoyam       112
16               Turmeric       111
17                 Barley       108
18               Plantain       106
19                  Mango       106
20             Watermelon       105
21        Pawpaw (Papaya)       102
22                  Wheat       101
23                    Yam        99
24                 Carrot        98
25  Garden Egg (Eggplant)        98
26                Cabbage        96
27    Scent  Leaf (Basil)        93
28             Grapefruit        93
29           Irish Potato        92
30                   Okra        90
31                 Orange        90
32                  Lemon        89
33   Fluted Pumpkin (Ugu)        83
34        Amaranth leaves        78
35            Castor Seed        73
36           Sweet Potato        71
37            oziza leave        69
38                 Ginger        67
39                Soybean        66
40         Cowpea (Beans)        65
41                Spinach        55
42                Cassava        44
43                  Onion        43
44             Bitterleaf        40
45                   Rice        32
46                 Cashew        31
47                 Pepper        29
48            Bambara Nut        20
49              Groundnut        13
50                 Millet        11

✅Observation/Insight: Maize emerged as the most demanded product, recording the highest number of orders (236 orders). This highlights its dominance in overall sales volume.

#### Products with the highest unique customers

In [406]:
#count of uniquecustmer_id per product 
product_sales.groupby('product_name')[['customer_id']].nunique().sort_values(by ='customer_id', ascending= False).reset_index()

product_name  customer_id
0                   Maize          226
1               Tangerine          193
2                Cucumber          177
3             Sesame Seed          160
4                  Tomato          153
5                 Avocado          148
6               Pineapple          144
7                    Lime          143
8                  Banana          138
9               Sunflower          136
10                  Guava          133
11        Celosia (Shoko)          131
12                Coconut          130
13              Waterleaf          118
14                Cocoyam          112
15                Lettuce          112
16               Turmeric          109
17                 Barley          106
18               Plantain          104
19             Watermelon          104
20                  Mango          102
21                  Wheat          100
22                 Carrot           98
23                    Yam           98
24        Pawpaw (Papaya)           97
25                Cabbage           96
26  Garden Egg (Eggplant)           95
27    Scent  Leaf (Basil)           92
28           Irish Potato           91
29             Grapefruit           90
30                  Lemon           89
31                   Okra           89
32                 Orange           88
33   Fluted Pumpkin (Ugu)           83
34        Amaranth leaves           77
35            Castor Seed           72
36           Sweet Potato           69
37            oziza leave           67
38                Soybean           66
39                 Ginger           65
40         Cowpea (Beans)           64
41                Spinach           54
42                Cassava           43
43                  Onion           43
44             Bitterleaf           40
45                   Rice           32
46                 Cashew           31
47                 Pepper           29
48            Bambara Nut           20
49              Groundnut           12
50                 Millet           11

✅Observation/Insight: Maize also emerged as the most preferred product, with 226 distinct customers purchasing it, reflecting broad appeal across the customer base.